python/pyarrow/tests/test_schema.py

```diff
@@ -23,6 +23,13 @@
 import pyarrow as pa
 
 
+def test_schema_constructor_errors():
+    with pytest.raises(ReferenceError):
+        repr(pa.Schema())
```

Comment:
I know you are just moving those tests around, but I wonder why `repr()` needs to be called here. Shouldn't the constructor itself raise an error?

Reply:
I'm just writing for Schema and DataTypes, it'll raise TypeError too